src/charm.py

```diff
@@ -218,11 +222,6 @@ def _restart(self, event: EventBase) -> None:
 
     def init_server(self):
         """Calls startup functions for server start."""
-        # don't run if leader has not yet created passwords
-        if not self.state.cluster.internal_user_credentials:
```

Comment:
Wouldn't this run and attempt to start the server even prior to having a relation with `opensearch` (and receiving kiabanserver)?

Reply:
`kibanasever` was never returned by `self.state.cluster.internal_user_credentials`.

It was only `CHARM_USERS` that contained `monitor` only. (Note `monitor` user never existed, we just "pretended" it's there by setting a `monitor-password` for it. Has never been created for real.)